pkgs/by-name/wc/wcurl/package.nix

```
@@ -0,0 +1,42 @@
+{ stdenv
+, lib
+, curl
+, fetchFromGitHub
+, makeWrapper
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wcurl";
+  version = "2024-07-07";
+
+  src = fetchFromGitHub {
+    owner = "Debian";
+    repo = "wcurl";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-qrYYPEgT17/7RbuiiOXnxAgi7TbMK/aAcMryTiQGZfo=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D wcurl $out/bin/wcurl
```

I would suggest using to avoid repeating binary name, and set the permissions (non blocking tho)
```suggestion
    install -Dm755 wcurl -t $out/bin
```

---

Why 755? I can see this is done elsewhere, but the 7 seems strange, as the permissions shouldn't be rwx.

---

I'm no sure, you might be right I guess `555` wins if this does not cause any issues

---

As far as I can tell, 755 is the default for `install`, but whether I specify it or not, the file ends up as r-xr-xr-x, so something must be removing the write bit. So I've changed it to use -t, but have left off the permissions for now, as there are plenty of examples that leave this off.

---

Seems like a good compromise

---

> As far as I can tell, 755 is the default for `install`, but whether I specify it or not, the file ends up as r-xr-xr-x, so something must be removing the write bit.

nix only tracks the executable bit permission, all other permissions are ignored: https://nix.dev/manual/nix/2.23/store/file-system-object